tests/e2e/specs/wcpay/shopper/shopper-checkout-free-cart.spec.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import config from 'config';
+
+const {
+	shopper,
+	merchant,
+	clearAndFillInput,
+	uiUnblocked,
+} = require( '@woocommerce/e2e-utils' );
+
+/**
+ * Internal dependencies
+ */
+import {
+	setupCheckout,
+	fillCardDetails,
+	fillCardDetailsWCB,
+} from '../../../utils/payments';
+import { shopperWCP, merchantWCP, checkPageExists } from '../../../utils';
+
+const productName = config.get( 'products.simple.name' );
+const billingDetails = config.get( 'addresses.customer.billing' );
+const card = config.get( 'cards.basic' );
+
+describe( 'Checkout with free Cart & after Cart updates on Checkout page', () => {
+	beforeAll( async () => {
+		try {
+			await checkPageExists( 'checkout-wcb' );
+		} catch ( error ) {
+			await merchant.login();
+			await merchantWCP.addNewPageCheckoutWCB();
+			await merchant.logout();
+		}
+	} );
+
+	beforeEach( async () => {
+		await shopper.goToShop();
+		await shopper.addToCartFromShopPage( productName );
+		await shopper.goToCart();
+		await clearAndFillInput( '#coupon_code', 'free' );
+		await expect( page ).toClick( 'button[name="apply_coupon"]' );
+	} );
+
+	afterAll( async () => {
+		// Clear the cart at the end so it's ready for another test
+		await shopperWCP.emptyCart();
+	} );
+
+	it( 'checkout a free cart', async () => {
+		await shopper.goToCheckout();
+		await shopper.fillBillingDetails( billingDetails );
+		await page.waitFor( 1000 );
```

Comment:
Would waiting for a selector be better here? This could lead to flakiness.

Reply:
Not sure, followed the same as the existing utility function `setupCheckout` in `utils/payments`. I cannot reuse the utility function as such, as I don't want the last line of the utility function.

Reply:
Could we create a seperate issue to fix this for both [`setupCheckout`](https://github.com/Automattic/woocommerce-payments/blob/5e61ad2629836ecd49583a2668862f6e2e6b9967/tests/e2e/utils/payments.js#L243) and this code?

Reply:
I noticed you're already using it below so something similar could be done here.
```
await page.waitForSelector( blocksPageEnabledSelector );
```
But not a blocker, so I'll let you take a call.